x-pack/plugins/security_solution/public/management/hooks/response_actions/use_get_automated_action_list.ts

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useQuery } from '@tanstack/react-query';
+import { lastValueFrom, of } from 'rxjs';
+
+import { mergeMap } from 'rxjs/operators';
+
+import { compact, map } from 'lodash';
+import type { PaginationInputPaginated } from '../../../../common/search_strategy/security_solution/response_actions/common';
+import { Direction } from '../../../../common/search_strategy/security_solution/response_actions/types';
+import type {
+  ActionResponsesRequestOptions,
+  ActionRequestOptions,
+  ActionRequestStrategyResponse,
+  ActionResponsesRequestStrategyResponse,
+} from '../../../../common/search_strategy/security_solution/response_actions';
+import { ResponseActionsQueries } from '../../../../common/search_strategy/security_solution/response_actions';
+import { useKibana } from '../../../common/lib/kibana';
+import type {
+  EndpointAutomatedActionListRequestQuery,
+  EndpointAutomatedActionResponseRequestQuery,
+} from '../../../../common/endpoint/schema/automated_actions';
+
+interface GetAutomatedActionsListOptions {
+  skip?: boolean;
+}
+
+export const useGetAutomatedActionList = (
+  query: EndpointAutomatedActionListRequestQuery,
+  { skip }: GetAutomatedActionsListOptions
+) => {
+  const { data } = useKibana().services;
+
+  const { alertIds } = query;
+  return useQuery(
+    ['get-automated-action-list', { alertIds }],
+    async () => {
+      const responseData = await lastValueFrom(
+        data.search.search<ActionRequestOptions, ActionRequestStrategyResponse>(
+          {
+            alertIds,
+            sort: {
+              direction: Direction.desc,
+              field: '@timestamp',
+            },
+            factoryQueryType: ResponseActionsQueries.actions,
+            pagination: generateTablePaginationOptions(0, 5),
```

Comment:
```suggestion
            pagination: generateTablePaginationOptions(0, 10),
```

Reply:
do we support pagination? if not let's set a bigger value and then come up with either pagination or an infinite scroll solution for that

Reply:
I will the code connected to pagination, let's keep it in mind and update with another iteration, alright?